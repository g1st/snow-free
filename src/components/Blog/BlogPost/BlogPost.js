import React from 'react';

import classes from './BlogPost.css';

const blogPost = props => (
  <div className={classes.Wrapper}>
    <div className={classes.Card}>
      <div className={classes.CardContent}>
        <div className={classes.CardInner}>
          <div className={classes.CardImage}>
            <img src={props.img} alt="Blog post" />
            <div className={classes.CardTag}>{props.author}</div>
          </div>
          <h1 className={classes.Heading}>Lorem Ipsum</h1>
          <span className={classes.Date}>August 14, 2018</span>
          <div className={classes.Paragraph}>{props.children}</div>
        </div>
      </div>
    </div>
  </div>
);

export default blogPost;
