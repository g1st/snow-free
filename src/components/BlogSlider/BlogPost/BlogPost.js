import React from 'react';

import classes from './BlogPost.css';

const blogPost = props => (
  <div className={classes.Card}>
    <div className={classes.CardContent}>
      <div className={classes.CardInner}>
        <div className={classes.CardImage}>
          <img src={props.img} alt="Blog post" />
          <div className={classes.CardTag}>{props.author}</div>
        </div>
        <h1 className={classes.Heading}>Lorem Ipsum</h1>
        <span className={classes.Date}>August 14, 2018</span>
        <p className={classes.Paragraph}>{props.children}</p>
      </div>
    </div>
  </div>
);

export default blogPost;
