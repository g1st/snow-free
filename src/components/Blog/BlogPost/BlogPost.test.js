import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import BlogPost from './BlogPost';

describe('BlogPost component', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = { img: 'image', author: 'John' };
    wrapper = shallow(<BlogPost {...props} />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should display author's name", () => {
    expect(wrapper.find('.CardTag').text()).toBe('John');
  });

  it('should display image', () => {
    expect(
      wrapper
        .find('.CardImage')
        .childAt(0)
        .props().src
    ).toBe('image');
  });
});
