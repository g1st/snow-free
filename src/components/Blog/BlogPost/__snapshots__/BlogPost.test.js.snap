// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost component matches snapshot 1`] = `
<div
  className="Wrapper"
>
  <div
    className="Card"
  >
    <div
      className="CardContent"
    >
      <div
        className="CardInner"
      >
        <div
          className="CardImage"
        >
          <img
            alt="Blog post"
            src="image"
          />
          <div
            className="CardTag"
          >
            John
          </div>
        </div>
        <h1
          className="Heading"
        >
          Lorem Ipsum
        </h1>
        <span
          className="Date"
        >
          August 14, 2018
        </span>
        <div
          className="Paragraph"
        />
      </div>
    </div>
  </div>
</div>
`;
